proxy/src/transparency/h1.rs

```diff
@@ -56,6 +64,10 @@ pub fn reconstruct_uri<B>(req: &mut http::Request<B>) -> Result<(), ()> {
             .expect("socket address is valid authority");
         set_authority(req.uri_mut(), auth);
 
+        // note that the request originally had no authority.
+        let _ = req.extensions_mut()
```

Comment:
As mentioned in Slack, instead of putting an extension here, this setting of the authority can instead just be done right before giving the `Client` instead. So, I'd suggest removing the code from `// last resort ... ` downwards.